src/model/OWL.plt

```diff
@@ -0,0 +1,172 @@
+:- begin_tripledb_tests(
+        'model_OWL',
+        'package://knowrob/owl/test/test_owl.owl',
+        [ namespace('http://knowrob.org/kb/test_owl#')]
+  ).
+
+:- use_module('./OWL.pl').
+
+:- use_module(library('lang/terms/is_a'),
+    [ subclass_of/2,
+      subproperty_of/2
+    ]).
+
+:- use_module(library('db/tripledb'),
+    [ tripledb_load/2,
+      tripledb_ask/3
+    ]).
+
+:- use_module(library('rostest')).
+
+test('is_class') :-
+  assert_true(is_class(test:'E')),
+  assert_true(is_class(owl:'Thing')),
+  assert_true(is_class(owl:'Nothing')),
+  assert_false(is_class(test:'DoesNotExists')),
+  assert_false(is_class(test:'p')),
+  assert_false(is_class(test:'bIndiv')),
+  assert_false(is_class(test:'P')).
+
+test('is_restriction/1') :-
+  assert_true((
+    is_restriction(Entity), 
+    tripledb_ask(Entity,owl:'onProperty',test:'s'),
+    tripledb_ask(Entity,owl:'allValuesFrom',test:'Range2')
+  )),
+  assert_true((
+    is_restriction(Entity), 
+    tripledb_ask(Entity,owl:'onProperty',test:'p'),
+    tripledb_ask(Entity,owl:'someValuesFrom',test:'B'),
+    % Additionally test if the restriction is the superclass of the correct class
+    subclass_of(test:'A2',Entity)
+  )),
+  % false cases
+  assert_false((
+    is_restriction(Entity), 
+    tripledb_ask(Entity,owl:'onProperty',test:'s'),
+    tripledb_ask(Entity,owl:'allValuesFrom',test:'Range1')
+  )),
+  assert_false((
+    is_restriction(Entity), 
+    tripledb_ask(Entity,owl:'onProperty',test:'t'),
+    tripledb_ask(Entity,owl:'allValuesFrom',test:'Range2')
+  )).
+
+test('is_restriction/2') :-
+  assert_true((subclass_of(test:'A',R), is_restriction(R,only(test:'s', test:'Range2')))),
```

Comment:
if it fails, the log output would not contain information about for which entity it failed.
Better make separate calls to assert_true